src/build.js

```diff
@@ -0,0 +1,43 @@
+/* @flow */
+import path from 'path';
+import {createWriteStream} from 'fs';
+import streamToPromise from 'stream-to-promise';
+
+import * as fs from './util/promised-fs';
+import {zipDir} from './util/zip-dir';
+
+
+export default function build({sourceDir, buildDir}: Object) {
```

Comment:
I thought I'd try Flow and see how annoying/helpful it is. I want to solve these problems:

* Unit tests are the fastest tests but they don't cover the implicit contracts between compositions of units
* Refactoring is hard with a suite of only unit tests
* Changing the interface of any function is hard in a dynamic language

Flow can solve all of these problems. It also has a nice escape hatch where you can annotate a type as `any`. I would like to focus on solving the aforementioned problems only so hopefully we can ignore typing when not necessary.

In this `build()` signature example, the `Object` is just a placeholder. I want to make it into a more meaningful type with: https://github.com/mozilla/web-ext/issues/39

As a counter example, we had a bug in `jpm` once where a refactoring changed the signature of all command handlers but since the unit tests only ran each command handler directly, it was uncaught. The alternative approach to this is to run every test as a shell command so that you invoke the entire process but this adds a magnitude of slowness (at scale) that is unacceptable to me.

By declaring *intent* of usage with type annotations you can catch many incorrect assumptions made by callers when they misunderstand the intent. Flow actually can do a lot of this without the need for explicit type annotations.